01_host/05_consensus/block_production.adoc

```diff
@@ -143,6 +150,53 @@ is required to produce a block. For the definitions of _Epoch-Randomness_
 (<<defn-epoch-randomness>>) and _VRF_ functions (<<sect-vrf>>).
 ****
 
+[#block-production-secondary-slots]
+===== Secondary Slots
+****
+**Secondary slots** work along side primary slot to ensure consistent block
+production, as described in <<sect-block-production-lottery>>. The secondary
+assignee of a block is determined by calculating a specific value, stem:[i_d],
+which indicates the index in the authority set (<<defn-authority-list>>). The
+corresponding authority in that set has the right to author a secondary block.
+
+[stem]
+++++
+p larr h("Enc"_("SC")(r,s))\
+i_d larr p mod A_l
+++++
+
+where:
+
+* stem:[r] is the Epoch randomness (<<defn-epoch-randomness>>).
+* stem:[s] is the slot number (<<defn-epoch-slot>>).
+* stem:["Enc"_("SC")(...)] encodes its inner value to the corresponding SCALE value.
+* stem:[h(...)] creates a 256-bit Blake2 hash from its inner value.
+* stem:[A_l] is the lengths of the authority list (<<defn-authority-list>>).
+
+To create the necessary VRF outputs, a transcript is construced (<<sect-vrf>>).
+
+[stem]
+++++
+t_1 larr "Transcript"("'BABE'")\
+t_2 larr "append"(t_1, "'slot number'", s)\
+t_3 larr "append"(t_2, "'current epoch'", e_i)\
+t_4 larr "append"(t_3, "'chain randomness'", r)\
+t_5 larr "append"(t_4, "'vrf-nm-pk'", p_k)\
+t_6 larr "meta-ad"(t_5, "'VRFHash'", "False")\
+t_7 larr "meta-ad"(t_6, 64_("le"), "True")\
+i larr "prf"(t_7, "False")\
+o = s_k * i\
+p larr sf "dleq_prove"(t_6, i)
+++++
+
+stem:[e_i] is the epoch index (<<defn-epoch-slot>>), stem:[s_k] is the private
+key and stem:[p_k] is the public key of the authority. stem:["dleq_prove"] is
+described in <<defn-vrf-dleq-prove>>. Values stem:[o] and stem:[p] are then used
+in the Pre-Digest item (<<defn-babe-header>>). In case of secondary slots with
+plain outputs, respectively the Pre-Digest being of value _2_, the transcript is
+skipped.
+****
```

Comment:
Now this completely breaks with the spec so far. If each validator uses its own private key, the output can not be the same for everybody anymore. Each validator would assume a different ordering, so this does not match with the ordering being known from the start. And VRF are used so the ordering is not revealed. 

I think it is more likely that this is just a second VRF slot, working just like the first one, but on different randomness. 

 At least that is how I understand this section. I have not looked into the code, but I would assume, the same threshold is used her as for the first slot to check if a slot is assigned.